Add Flathub plugin

ShadowApex opened the pull request:
This PR adds the FlatHub plugin to the Crankshaft plugin store. It allows you to browse, search, install, and uninstall Flatpaks from Flathub and also creates Steam shortcuts using the [steam-shortcut-manager](https://github.com/ShadowBlip/steam-shortcut-manager) tool (written in Go). Using the shortcut manager it will also try and download the grid artwork for a given Flatpak automatically so Flatpaks look nice in your library.

![](https://github.com/ShadowBlip/crankshaft-flathub/raw/main/doc/image01.png)

![](https://github.com/ShadowBlip/crankshaft-flathub/raw/main/doc/image02.png)

**NOTE:** This current release only allows you to search using a keyboard. I'm hoping to add on-screen keyboard support in future releases.

ShadowApex committed:
Add Flathub plugin

coolavery commented:
Thanks for the submission! Looks great, only feedback is to see if you can add a description, just to make it more clear what the plugin does in the plugin store before installing it. This can be added in your `plugin.toml`, there's an example here: https://git.sr.ht/~avery/proton-tools/tree/main/item/plugin.toml

ShadowApex committed:
Update SHA for re-release of Flathub v1.0.0 with plugin store description

ShadowApex commented:
@coolavery Thanks! I've updated `plugin.toml` with a description here:
https://github.com/ShadowBlip/crankshaft-flathub/blob/main/plugin.toml#L22

I created a new release and updated the SHA of this PR to reflect the new tar.gz with the updated `plugin.toml`.
